### ISSUE TITLE
Fix Request for HttpClientBuilderWithoutTimeouts: a false positive

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#125]
The following code has a false positive, timeouts are set on a RequestConfig, which is set on a HttpClient, set on the HttpRequestFactory.

```
    RestTemplate getRestTemplate(final Integer maxConnections, final Integer maxConnectionsPerHost,
                                 final Integer connectionTimeoutInMilliSeconds, final Integer readTimeoutInMilliSeconds, final SSLContext sslContext) {

        PoolingHttpClientConnectionManager poolingHttpClientConnectionManager = sslContext != null ?
                getPoolingHttpClientConnectionManagerWithSslEnabled(sslContext, maxConnections, maxConnectionsPerHost) :
                getPoolingHttpClientConnectionManagerWithoutSslEnabled(maxConnections, maxConnectionsPerHost);

        RequestConfig requestConfig = RequestConfig.custom()
                .setConnectTimeout(connectionTimeoutInMilliSeconds)
                .setConnectionRequestTimeout(connectionTimeoutInMilliSeconds)
                .setSocketTimeout(readTimeoutInMilliSeconds)
                .build();

        HttpClient httpClient = HttpClients.custom()
                .setDefaultRequestConfig(requestConfig)
                .setConnectionManager(poolingHttpClientConnectionManager)
                .disableRedirectHandling()
                .disableConnectionState()
                .build();

        try {
            HttpComponentsClientHttpRequestFactory httpComponentsClientHttpRequestFactory = new  HttpComponentsClientHttpRequestFactory();  // false positive
            httpComponentsClientHttpRequestFactory.setHttpClient(httpClient);
            return new RestTemplate(httpComponentsClientHttpRequestFactory);
        } catch (Exception exception) {
            throw new RefundTechnicalException("Error while creating Http Connection Factory");
        }
    }
```

**Comment by stokpop** [comment]
Created a fix in HttpClientBuilderWithoutTimeouts branch
